website/docs/connect-data/reference/open-ai.md

```diff
@@ -0,0 +1,222 @@
+
+
+## Connect OpenAI
+
+ <figure>
+  <img src="/img/open-ai-main2.png" style= {{width:"810px", height:"auto"}} alt="Server-side Filtering on Table"/>
+  <figcaption align = "center"><i>OpenAI datasource.</i></figcaption>
+</figure>
+
+
+### Connection parameters
+
+The following section is a reference guide that provides a complete description of all the parameters to connect to a OpenAI datasource.
+
+
+#### Authentication type
+
+<dd>
+
+Authentication Type refers to the method used to verify the identity of users or systems interacting with the OpenAI API. 
+
+*Options:*
+
+* **Bearer Token:** It is a type of access token that is included in the request headers to authenticate and authorize API requests.
+
+
+</dd>
+
+
+#### API Key
+
+<dd>
+
+The OpenAI uses API keys for authentication. Visit the [API Keys](https://platform.openai.com/account/api-keys) page to retrieve the API key.
+
+</dd>
+
+
+## Query OpenAI
+
+The following section is a reference guide that provides a description of the available commands with their parameters to create OpenAI queries.
+
+### Chat
+
+The Chat command generates human-like text based on input prompts. The following section lists all the available parameters:
+
+  <figure>
+  <img src="/img/open-ai-chat2.png" style= {{width:"720px", height:"auto"}} alt="Server-side Filtering on Table"/>
+  <figcaption align = "center"><i>OpenAI | Chat command</i></figcaption>
+</figure>
+
+
+#### Models
+
+<dd>
+
+It refers to the pre-trained language models provided by OpenAI. You can select from the available list of models, including options like GPT-3, GPT-4, and others.
+
+</dd>
+
+
+#### Temperature
+
+<dd>
+
+ Temperature determines the level of randomness in the output. It ranges between 0 and 2. 
+ 
+ Lower values for temperature result in more consistent outputs (e.g. 0.2), while higher values generate more diverse and creative results (e.g. 1.0). Select a temperature value based on the desired trade-off between coherence and creativity for your specific application.
+
+
+</dd>
+
+#### Messages
+
+<dd>
+
+Messages serve as input interactions between the user and the model. The roles include two types: **user** and **system**. You can create multiple messages of each type to make your conversation just the way you want. 
+
+* **user**: Input provided by the user to instruct or guide the model. For example, if you are using an Input widget to enter the prompt, you can use `{{Input1.text}}`.
+
+* **system**: The system message serves as a means to provide additional context, set guidelines, or convey the overall objective of the task. It helps shape the behavior of the model's responses. For example, you can use the system message to give personality to the responses or add task-specific instructions. For example, you can set system as:
```

Comment:
System messages are usually added before the user message. Can you reorder this?